shpkpr/commands/cmd_bluegreen.py

```diff
@@ -46,44 +60,63 @@ def cli(logger, marathon_client, marathon_lb_url, initial_instances, max_wait,
     for app in rendered_templates:
         validate_app(app)
 
+        # fetch all previous deploys from marathon and abort if there is more
+        # than one stack currently active.
         previous_deploys = fetch_previous_deploys(marathon_client, app)
-
         if len(previous_deploys) > 1:
-            return safe_resume_deploy(logger,
-                                      force,
-                                      max_wait,
-                                      step_interval,
-                                      initial_instances,
-                                      marathon_client,
-                                      marathon_lb_url,
-                                      previous_deploys)
-
+            raise DualStackAlreadyExists("Both blue and green apps detected")
+        # transform the app to be deployed to apply the correct labels and
+        # ID-change that will allow marathon-lb to cut traffic over as necessary.
         new_app = prepare_deploy(previous_deploys, app, initial_instances)
 
         logger.log('Final App Definition:')
-        logger.log(_pretty_print(new_app))
+        logger.log(json.dumps(new_app, indent=4, sort_keys=True))
         if force or click.confirm("Continue with deployment?"):
-            marathon_client.deploy([new_app]).wait()
-
-            if len(previous_deploys) == 0:
-                # This was the first deploy, nothing to swap
-                return True
-            else:
-                # This is a standard blue/green deploy, swap new app with old
-                old_app = select_last_deploy(previous_deploys)
-                return swap_bluegreen_apps(logger,
-                                           force,
-                                           max_wait,
-                                           step_interval,
-                                           initial_instances,
-                                           marathon_client,
-                                           marathon_lb_url,
-                                           new_app,
-                                           old_app,
-                                           time.time())
-
-
-def _pretty_print(dict):
-    """Pretty print a dict as a json structure
+            try:
+                deploy_and_swap(marathon_client,
+                                new_app,
+                                previous_deploys,
+                                logger,
+                                force,
+                                max_wait,
+                                step_interval,
+                                initial_instances,
+                                marathon_lb_url)
```

Comment:
NAB: But I'd love to pair with you at some stage in getting the amount of things passed in here down.